### PR TITLE
Adding a optional disclaimer to the anonymous upload page

### PR DESCRIPTION
**Pull request by MorrisJobke**
We got a lot of requests how to add a disclaimer text to the upload page. This adds a really simple way to show a disclaimer right on the side:

bildschirmfoto 2016-09-08 um 09 09 26
- [x] add a setting to the sharing admin section to enter this disclaimer
- [ ] writing a documentation page about how to add a disclaimer text

@nextcloud/designers Feedback about the design is very welcome

cc @LukasReschke

### CODE REVIEW

**Comment by MorrisJobke** [comment]
cc @karlitschek ;)

File: apps/files_sharing/lib/Controllers/ShareController.php [hunk 1]
```diff
@@ -357,6 +357,9 @@ public function showShare($token, $path = '') {
 		$shareTmpl['previewEnabled'] = $this->config->getSystemValue('enable_previews', true);
 		$shareTmpl['previewMaxX'] = $this->config->getSystemValue('preview_max_x', 1024);
 		$shareTmpl['previewMaxY'] = $this->config->getSystemValue('preview_max_y', 1024);
+		$shareTmpl['disclaimer'] = 'Disclaimer: Nihil voluptas voluptas quia culpa. In quia doloribus molestiae explicabo. Nihil totam omnis et. Labore quia aut omnis aut vel.
```
Comment (nickvergessen) [inline apps/files_sharing/lib/Controllers/ShareController.php]: :1234:

Comment (MorrisJobke) [inline apps/files_sharing/lib/Controllers/ShareController.php]: That's the reason it is still a WIP ;)

Comment (MorrisJobke) [inline apps/files_sharing/lib/Controllers/ShareController.php]: I just want to collect feedback earlier 😉

Comment (LukasReschke) [inline apps/files_sharing/lib/Controllers/ShareController.php]: Will add the settings myself.

**Comment by LukasReschke** [comment]
Considering also other customer requests and the ability to upload files as well via the generic sharing public link, I'm thinking whether we can also add this somewhere ad the bottom of normal sharing dialogs?

This pretty much comes up with a lot of customers as a legal requirement and having them to monkey patch templates is very error-prone.

**Comment by LukasReschke** [comment]
Maybe we could to the other pages add something like a small `| Disclaimer` after the slogan and clicking on that opens a modal showing the chosen text. Or makes the text appear on top of the footer or whatsoever.

Then again, I think there might be two use-cases. One having a general long disclaimer (which the slogan link would be nice for), and one having this small kind of text like "Uploading sensitive data is prohibited. Blah blah blah".

Thoughts? – I think for now it's ok just to address the warning for sharing links. But if we can think of a solution that also applies to the other sharing views, that would be nice.

Another possibility would be to make the disclaimer basically pop-up per modal on all sharing views with an "Accept" button. If the user clicks on accept, this is stored via cookie and the dialog isn't displayed anymore. This is what some banking solutions do, far away from a perfect user experience though :)

**Comment by karlitschek** [comment]
sweet 👍

**Comment by MorrisJobke** [comment]
I will continue on this as discussed with lukas. FOr now I will only introduce the disclaimer to the public shares with hidden file list. If more is requested we can extend this afterwards.

**Comment by MorrisJobke** [comment]
I added it now to the settings page of the sharing settings:

bildschirmfoto 2016-09-08 um 18 25 32
bildschirmfoto 2016-09-08 um 18 25 52

**Comment by MorrisJobke** [comment]
Ready for review :)

I even tested with Edge :)

File: settings/templates/admin/sharing.php [hunk 2]
```diff
@@ -104,4 +104,12 @@
 			<?php if ($_['allowShareDialogUserEnumeration'] === 'yes') print_unescaped('checked="checked"'); ?> />
 		<label for="shareapi_allow_share_dialog_user_enumeration"><?php p($l->t('Allow username autocompletion in share dialog. If this is disabled the full username needs to be entered.'));?></label><br />
 	</p>
+	<p>
+		<input type="checkbox" id="publicShareDisclaimer" class="checkbox noJSAutoUpdate"
+			<?php if ($_['publicShareDisclaimerText'] !== null) print_unescaped('checked="checked"'); ?> />
+		<label for="publicShareDisclaimer"><?php p($l->t('Show disclaimer text on the public link upload page. (Only shown when the file list is hidden.)'));?></label>
+		<span id="publicShareDisclaimerStatus" class="msg" style="display:none"></span>
+		<br/>
+		<textarea placeholder="This text will be shown on the public link upload page when the file list is hidden." id="publicShareDisclaimerText" <?php if ($_['publicShareDisclaimerText'] === null) { print_unescaped('class="hidden"'); } ?>><?php p($_['publicShareDisclaimerText']) ?></textarea>
```
Comment (LukasReschke) [inline settings/templates/admin/sharing.php]: I guess that one should use L10n? I'll push that.

**Comment by LukasReschke** [comment]
Awesome stuff 👍 🚀 🎉

**Comment by MariusBluem** [comment]
Works also with Safari 😉 👍 ...Drone is failing 😕 @MorrisJobke

**Comment by LukasReschke** [comment]
I'll fix the failing unit test ;)

**Comment by MorrisJobke** [comment]
Thanks @LukasReschke 👍 to your changes

**Pull request merged**